packages/mc-scripts/src/commands/build-vite.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+const fs = require('fs-extra');
+const path = require('path');
+const { build } = require('vite');
+const pluginGraphql = require('@rollup/plugin-graphql');
+const pluginReact = require('@vitejs/plugin-react').default;
+const { useDynamicPublicPath } = require('vite-plugin-dynamic-publicpath');
+const {
+  generateTemplate,
+  htmlScripts,
+} = require('@commercetools-frontend/mc-html-template');
+const {
+  packageLocation: applicationStaticAssetsPath,
+} = require('@commercetools-frontend/assets');
+const paths = require('../config/paths');
+
+const DEFAULT_PORT = parseInt(process.env.HTTP_PORT, 10) || 3001;
+
+const pluginCustomApplication = () => {
+  return {
+    name: 'custom-application',
+    /**
+     * @type {import('vite').IndexHtmlTransformHook}
+     */
+    transformIndexHtml(rawHtml, _ctx) {
+      // Ensure to use the `cdnUrl` value when loading the entry point.
+      // NOTE: with Webpack this is done by setting the `__webpack_public_path__`.
+      const html = rawHtml.replace(
+        new RegExp(`<script type="module"(.*) src="(.*)">`, 'g'),
```

Comment:
I guess alternatively we could parse the HTML which might be more expensive. Maybe defining `function prependCdnUrlToScriptSrc` would help. At least I got stuck here deducting that :)